### ISSUE TITLE
No loss return during model training

### ISSUE THREAD

**Issue by Karenou** [havakv/pycox#48]
Hi, I encounter a problem when using LogisticHazard and DeepHitSingle model `to` fit a customer churn model.  

The dataset after performing label transformation is as below. There are around 1400 features and the duration ranges from 0 to 60. The dataset is imbalanced. The ratio of censoring data (event = 0) to remaining data is 5 in the training set and 20 in the eval set. I discretize it by setting num_duration equal 3.
![image](https://user-images.githubusercontent.com/44489747/91429847-5f00a080-e891-11ea-98f9-aa22b98c267a.png)

But when I try to fit the model, it just shows the training time per epoch, but no training and eval loss is shown.  May I know how to solve the problem? Many thanks.
`
log = model.fit(x_train, y_train, batch_size, epochs, callbacks, val_data=val_sample)
`
![image](https://user-images.githubusercontent.com/44489747/91428149-f44f6500-e890-11ea-8629-97a50596ecb2.png)

**Comment by Karenou** [comment]
I have solved the problem by filling the null values in the data. Thanks!

**Comment by havakv** [comment]
I'm glad you figured this out! Were there null values in the covairates `x_train` or in the labels `y_train`? If the problems were caused by null values in the `y_train` we should probably fix the discretisation code or maybe raise an appropriate warning/error

**Comment by Karenou** [comment]
Thanks for your quick response. There were null values in the covariates `x_train`. 

I have encountered another problem which is specific to my dataset. Since the dataset I am using is very imbalanced, I undersample the negative class (event = 0) in the training set but not the validation set. But when I use the negative log-likelihood loss, eg. `NLLMTLRLoss`, the validation loss is far less than the training loss.  Is there another way to add customized evaluation metrics in model.fit for model training aside from the negative log-likelihood? Or I just write another `Loss` class considering the class weight?

Many thanks!

**Comment by havakv** [comment]
So, I don't really want to check if the covariates `x_train` are "legal", as they are passed though an arbitrary pytorch net and could potentially serve a purpose there. Also, the convention in pytorch seem to not check the loss function for "leagal" values, so I think it makes sense to close this issue unless you disagree with this decision, and if it is a reoccurring problem for others, we can revisit this later?
